### PR TITLE
3763 Enhance the doc of `ThreadDataLoader` for `num_workers`

### LINKED ISSUE #3763: Clarify the `num_workers` in `ThreadDataLoader`
**Is your feature request related to a problem? Please describe.**
When I was introducing GPU transforms and the associated `ThreadDataLoader` to users, got several times feedback about the `num_workers` arg, which is confusing that users think it means the multi-threads in `ThreadDataLoader`, but actually it's the multi-processing workers of PyTorch DataLoader.
Would be nice to clarify this arg and the use cases.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #3763 .

### Description
This PR enhanced the doc of `ThreadDataLoader` to clarify `num_workers` arg and the typical usage.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/data/thread_buffer.py [hunk 1]
```diff
@@ -100,7 +106,11 @@ class ThreadDataLoader(DataLoader):
         buffer_size: number of items to buffer from the data source.
         buffer_timeout: time to wait for an item from the buffer, or to wait while the buffer is full when adding items.
         num_workers: number of the multi-processing workers in PyTorch DataLoader.
+            note that it's not the number of threads for the `ThreadBuffer` in this class, which always uses 2 threads:
```
Comment (wyli) [inline monai/data/thread_buffer.py:109]: I don't see any good use case with num_workers>0, perhaps we just remove it?

Comment (Nic-Ma) [reply monai/data/thread_buffer.py]: Makes sense to me.
I removed it in the latest commit and added some description in `kwargs`.

Thanks.

Comment (wyli) [reply monai/data/thread_buffer.py]: Thanks, I think after this, we don't need to mention num_workers in the docstring to avoid confusion, correct?

Comment (Nic-Ma) [reply monai/data/thread_buffer.py]: OK, updated in the latest commit.

Thanks.

**Review by wyli — APPROVED** [review]
Thanks, it looks good to me.

**Pull request merged**